pkg/controller/node/rate_limited_queue.go

```diff
@@ -221,7 +219,7 @@ func (q *RateLimitedTimedQueue) Try(fn ActionFunc) {
 			val.ProcessAt = now.Add(wait + 1)
 			q.queue.Replace(val)
 		} else {
-			q.queue.RemoveFromQueue(val.Value)
+			q.queue.Remove(val.Value)
```

Comment:
The bug you describe is a real one, but you should just call remove on observing Pod deletion. This code is necessary to export correct number of evictions. Without it we'll be processing the same Pod over and over again until the Node won't come back. This is a level that does deduplication.